### PR TITLE
Overhaul README to be client-centric 

### PR DESCRIPTION
**Pull request by douglascamata**
The development section of the README should be focussed on developers. There will be links between this README and the public docs whenever they are ready to be published.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,11 +72,15 @@ This job will execute the shell script [k8s-webhook-cert-manager/generate_certif
 
 If you wish to learn more about TLS certificates management inside Kubernetes, check out [the official documentation for Managing TLS Certificate in a Cluster](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#create-a-certificate-signing-request-object-to-send-to-the-kubernetes-api).
 
-#### Manual management
+### Manual certificate management
```
Comment (douglascamata) [inline README.md]: Shall we start calling this option `Custom certificate management` to match with the yaml that is named `custom-certs`?

File: README.md [hunk 2]
```diff
@@ -11,53 +11,44 @@ New Relic requires the following environment variables to identify Kubernetes ob
...
+- `newrelic-metadata-injection-cfg`, registering the `newrelic-metadata-injection-svc` service as a MutatingAdmissionWebhook with the Kubernetes API.
 
-#### Automatic management
+### Automatic certificate management
 
 The certificate management can be automatic, using the Kubernetes extension API server (recommended, but optional):
 
 ```bash
-$ kubectl apply -f deploy/job.yaml
+$ kubectl apply -f http://download.newrelic.com/infrastructure_agent/integrations/k8s-metadata-injection-latest.yaml
```
Comment (areina) [inline README.md]: This is not correct. The user has to edit the file and update the value of the `clusterName` environment variable.

**Comment by douglascamata** [comment]
@areina I can see it being more simple if we move all the explanations to the end. Though as a customer I would like to explanation of what I'm going to do before I run the commands. I think we can assume that the client, in this case, has some k8s knowledge. Considering this, I don't think it's complicated to understand. It does have a lot of information before installation, but that was intentional.

Imagine a client reading the public docs, and after installing the default solution he discovers he actually wants the namespace-based solution. This is very frustrating and wastes time.

For example, the [APM section of our docs](https://docs.newrelic.com/docs/apm/new-relic-apm/getting-started/introduction-new-relic-apm) first explains many things about APM to speak about installation steps at the penultimate section.

Even the [.NET agent page](https://docs.newrelic.com/docs/agents/net-agent/installation/net-agent-install-introduction) follows a similar approach.

**Comment by douglascamata** [comment]
Also remember that this will be split into multiple pages, like a `Getting started` section with some items inside, then later a section for the installation with a point for auto cert, another one for manual cert, one explaining how to install/use the namespace-based option

File: README.md [hunk 3]
```diff
@@ -11,53 +11,52 @@ New Relic requires the following environment variables to identify Kubernetes ob
 - `NEW_RELIC_METADATA_KUBERNETES_POD_NAME`
 - `NEW_RELIC_METADATA_KUBERNETES_CONTAINER_NAME`
 
-These environment variables can be set manually by the customer, or they can be automatically injected using a MutatingAdmissionWebhook.
-New Relic provides an easy method for deploying this automatic approach.
+These environment variables can be set manually by the customer, or they can be automatically injected using a `MutatingAdmissionWebhook`.
```
Comment (smoya) [inline README.md]: I feel the introduction located in [docs/performance.md](docs/performance.md) is easy and simpler: 

> The injection of New Relic APM Metadata is implemented as a webhook using Kubernetes MutatingAdmissionWebhook, which is in fact an AdmissionWebhook.
>
>Admission webhooks run as part of the Kubernetes API Request Lifecycle.
>
>An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized.
>
>--- kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-are-they

Comment (smoya) [inline README.md]: Suggestion:

```
These environment variables can be set manually by the customer, However, this manual process can be very counterproductive. 

New Relic provides an easy method for automatically inject those environment variables in any Kubernetes pod container.
```

File: README.md [hunk 4]
```diff
@@ -11,53 +11,52 @@ New Relic requires the following environment variables to identify Kubernetes ob
...
 
-## Setup
+New Relic provides an easy method for deploying this automatic approach.
```
Comment (smoya) [inline README.md]: See https://github.com/newrelic/k8s-metadata-injection/pull/34/files#r253957240

**Comment by douglascamata** [comment]
Closing this PR in favor of a Google Docs because it gives a better overview of the documentation structure.

**Comment by douglascamata** [comment]
Closed in favor o #46

**Pull request closed without merging**